schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,370 @@
+      "properties": {
+        "default": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string"
+        },
+        "description": {
+          "type": "object",
+          "additionalProperties": {
+            "not": {
+              "type": "object"
+            }
+          }
+        },
+        "validators": {
+          "type": "array",
+          "items": {"$ref": "#/definitions/config_validator"}
+        }
+      },
+      "required": ["description"]
+    },
+    "config_validator": {
+      "anyOf": [
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "required"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type"]
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "regex"
+            },
+            "value": {
+              "type": "string"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type", "value"]
+        },
+        {
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "type": {
+              "const": "max_length"
+            },
+            "value": {
+              "type": "integer"
+            },
+            "description": {
+              "type": "object",
+              "additionalProperties": {
+                "not": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "required": ["type", "value"]
+        }
+      ]
+    },
+    "input_env_variables": {
+      "type": "string",
+      "enum": [
+        "USER_EMAIL",
+        "ENROLLMENT_IDENTIFIER",
+        "ENROLLMENT_USERNAME",
+        "ENROLLMENT_GROUP_IDENTIFIER",
+        "ENROLLMENT_DEVICE_UDID",
+        "DEVICE_IDENTIFIER",
+        "DEVICE_MODEL",
+        "DEVICE_NAME",
+        "DEVICE_SYSTEM_NAME",
+        "DEVICE_SYSTEM_VERSION",
+        "DEVICE_UI_IDIOM"
+      ]
+    },
+    "http_methods": {
+      "type": "string",
+      "enum": [
+        "DELETE",
+        "GET",
+        "HEAD",
+        "OPTIONS",
+        "PATCH",
+        "POST",
+        "PUT",
+        "TRACE"
+      ]
+    },
+    "connector_action_keys": {
+      "type": "string",
+      "enum": [
+        "DIRECT",
+        "USER_INPUT",
+        "OPEN_IN"
+      ]
+    },
+    "connector_action_user_input_formats": {
+      "type": "string",
+      "enum": [
```

Comment:
I don't think I like these enums being official in the schema.  It's going to be a very rigid contract now.

Also, it doesn't even support the right list of things (ex. the Boxer client code supports "range", "checkbox", "radio", etc.)

Reply:
That seems to be unnecessary forcing in the schema, even if its right w.r.t Boxer client.
Removed it.